Sources/StreamChat/Query/ChannelListQuery.swift

```diff
@@ -58,67 +58,67 @@ public enum InviteFilterValue: String, FilterValue {
 public extension FilterKey where Scope: AnyChannelListFilterScope {
     /// A filter key for matching the `cid` value.
     /// Supported operators: `in`, `equal`
-    static var cid: FilterKey<Scope, ChannelId> { "cid" }
+    static var cid: FilterKey<Scope, ChannelId> { .init(rawValue: "cid", keyPathValueProvider: { "cid" }) }
 
     /// A filter key for matching the `id` value.
     /// Supported operators: `in`, `equal`
     /// - Warning: Querying by the channel Identifier should be done using the `cid` field as much as possible to optimize API performance.
     /// As the full channel ID, `cid`s are indexed everywhere in Stream database where `id` is not.
-    static var id: FilterKey<Scope, String> { "id" }
+    static var id: FilterKey<Scope, String> { .init(rawValue: "id", keyPathValueProvider: { "cid" }) }
 
     /// A filter key for matching the `name` value.
-    static var name: FilterKey<Scope, String> { "name" }
+    static var name: FilterKey<Scope, String> { .init(rawValue: "name", keyPathValueProvider: { "name" }) }
 
     /// A filter key for matching the `image` value.
-    static var imageURL: FilterKey<Scope, URL> { "image" }
+    static var imageURL: FilterKey<Scope, URL> { .init(rawValue: "image", keyPathValueProvider: { "imageURL" }) }
```

Comment:
In the initial implementation, we used something like this: `#keyPath(ChannelDTO.name)` to avoid committing mistakes, is there any reason we are not following this approach?